tiledb/api/src/array/enumeration.rs

```diff
@@ -0,0 +1,502 @@
+                column.offsets().len() as u64,
+                &mut c_new_enmr,
+            )
+        };
+
+        if res == ffi::TILEDB_OK {
+            Ok(Enumeration {
+                context: self.context,
+                raw: RawEnumeration::Owned(c_new_enmr),
+            })
+        } else {
+            Err(self.context.expect_last_error())
+        }
+    }
+}
+
+impl<'data> TryAsColumn<'data> for Enumeration<'data> {
+    fn try_as_column(&'data self) -> TileDBResult<Column<'data>> {
+        let dtype = self.datatype()?;
+        let cell_val_num = self.cell_val_num()?;
+        let data = self.data()?;
+        let offsets = self.offsets()?;
+
+        let num_values = if offsets.is_empty() {
+            data.len() as u64 / (dtype.size() * cell_val_num as u64)
+        } else {
+            offsets.len() as u64
+        };
+
+        let value_size = if cell_val_num == u32::MAX {
+            dtype.size()
+        } else {
+            dtype.size() * cell_val_num as u64
+        };
+
+        Ok(Column::from_references(
+            num_values, value_size, data, offsets,
+        ))
+    }
+}
+
+impl<'ctx> Debug for Enumeration<'ctx> {
+    fn fmt(&self, f: &mut Formatter) -> FmtResult {
+        let name = self.name().map_err(|_| fmt::Error)?;
+        let dtype = self
+            .datatype()
+            .map_err(|_| fmt::Error)?
+            .to_string()
+            .unwrap_or("<unknown datatype>".to_owned());
+        let cell_val_num = self.cell_val_num().map_err(|_| fmt::Error)?;
+        let ordered = self.ordered().map_err(|_| fmt::Error)?;
+
+        // TODO: Add enumeration values display.
+
+        let json = json!({
+            "name": name,
+            "datatype": dtype,
+            "cell_val_num": cell_val_num,
+            "ordered": ordered
+        });
+        write!(f, "{}", json)
+    }
+}
+
+impl<'c1, 'c2> PartialEq<Enumeration<'c2>> for Enumeration<'c1> {
+    fn eq(&self, other: &Enumeration<'c2>) -> bool {
+        let names_match = match (self.name(), other.name()) {
+            (Ok(mine), Ok(theirs)) => mine == theirs,
+            _ => false,
+        };
+        if !names_match {
+            return false;
+        }
+
+        let types_match = match (self.datatype(), other.datatype()) {
+            (Ok(mine), Ok(theirs)) => mine == theirs,
+            _ => false,
+        };
+        if !types_match {
+            return false;
+        }
+
+        let cell_val_num_match =
+            match (self.cell_val_num(), other.cell_val_num()) {
+                (Ok(mine), Ok(theirs)) => mine == theirs,
+                _ => false,
+            };
+        if !cell_val_num_match {
+            return false;
+        }
+
+        let ordered_match = match (self.ordered(), other.ordered()) {
+            (Ok(mine), Ok(theirs)) => mine == theirs,
+            _ => false,
+        };
+        if !ordered_match {
+            return false;
+        }
+
+        // TODO: Match data and offsets
```

Comment:
Probably can fetch the column and `impl PartialEq` for the column, that would be a good way to do it.

Reply:
Yep. I wrote these before Column thinking I was gonna go fill out the `arbitrary_enumeration` implementation first. But then it bugged me I wasn't asserting the major differences so went back to figuring out the Column stuff. I hadn't thought of adding PartialEq on the Column itself. That's a great idea. My current plan is to just make sure that the sizes are equal and then figure Rust's idea of `std::memcmp`.